pkg/cmd/logs/tail.go

```diff
@@ -0,0 +1,185 @@
+type TailCmd struct {
+	apiBaseURL string
+	cfg        *config.Config
+	Cmd        *cobra.Command
+	format     string
+	LogFilters *logTailing.LogFilters
+	noWSS      bool
+}
+
+// NewTailCmd creates and initializes the tail command for the logs package
+func NewTailCmd(config *config.Config) *TailCmd {
+	tailCmd := &TailCmd{
+		cfg:        config,
+		LogFilters: &logTailing.LogFilters{},
+	}
+
+	tailCmd.Cmd = &cobra.Command{
+		Use:   "tail",
+		Args:  validators.NoArgs,
+		Short: "Listens for API request logs sent from Stripe to help test your integration.",
+		Long: fmt.Sprintf(`
+The tail command lets you tail API request logs from Stripe.
+The command establishes a direct connection with Stripe to send the request logs to your local machine.
+
+Watch for all request logs sent from Stripe:
+
+  $ stripe logs tail`),
+		RunE: tailCmd.runTailCmd,
+	}
+
+	tailCmd.Cmd.Flags().StringVar(
+		&tailCmd.format,
+		"format",
+		"",
+		`Specifies the output format of request logs
+Acceptable values:
+	'JSON' - Output logs in JSON format`,
+	)
+
+	// Log filters
+	tailCmd.Cmd.Flags().StringSliceVar(
+		&tailCmd.LogFilters.FilterAccount,
+		"filter-account",
+		[]string{},
+		`*CONNECT ONLY* Filter request logs by source and destination account
+Acceptable values:
+	'connect_in'  - Incoming connect requests
+	'connect_out' - Outgoing connect requests
+	'self'        - Non-connect requests`,
+	)
+	tailCmd.Cmd.Flags().StringSliceVar(&tailCmd.LogFilters.FilterIPAddress, "filter-ip-address", []string{}, "Filter request logs by ip address")
+	tailCmd.Cmd.Flags().StringSliceVar(
+		&tailCmd.LogFilters.FilterHTTPMethod,
+		"filter-http-method",
+		[]string{},
+		`Filter request logs by http method
+Acceptable values:
+	'GET'    - HTTP get requests
+	'POST'   - HTTP post requests
+	'DELETE' - HTTP delete requests`,
+	)
+	tailCmd.Cmd.Flags().StringSliceVar(&tailCmd.LogFilters.FilterRequestPath, "filter-request-path", []string{}, "Filter request logs by request path")
+	tailCmd.Cmd.Flags().StringSliceVar(
+		&tailCmd.LogFilters.FilterRequestStatus,
+		"filter-request-status",
+		[]string{},
+		`Filter request logs by request status
+Acceptable values:
+	'SUCCEEDED' - Requests that succeeded (status codes 200, 201, 202)
+	'FAILED'    - Requests that failed`,
+	)
+	tailCmd.Cmd.Flags().StringSliceVar(
+		&tailCmd.LogFilters.FilterSource,
+		"filter-source",
+		[]string{},
+		`Filter request logs by source
+Acceptable values:
+	'API'       - Requests that came through the Stripe API
+	'DASHBOARD' - Requests that came through the Stripe Dashboard`,
+	)
+	tailCmd.Cmd.Flags().StringSliceVar(&tailCmd.LogFilters.FilterStatusCode, "filter-status-code", []string{}, "Filter request logs by status code")
+	tailCmd.Cmd.Flags().StringSliceVar(
+		&tailCmd.LogFilters.FilterStatusCodeType,
+		"filter-status-code-type",
+		[]string{},
+		`Filter request logs by status code type
+Acceptable values:
+	'200' - All 200 status codes
```

Comment:
Should these be "All 2xx status codes", 4xx, 5xx or is it specifically 200, 400, and 500?

Reply:
2XX, 4XX, 5XX. I'll update the comment to make it more clear. Should we also make it so the user specifies `2XX` instead of `200`?